gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -332,9 +333,9 @@ public IGitCoreHeadSnapshot deriveHead() throws GitCoreException {
       // This might NOT necessarily be OK from the perspective of remote tracking status
       // i.e. the number of commits ahead of/behind remote, but in case of criss-cross histories
       // it's basically impossible to get these numbers correctly in a unambiguous manner.
-      @Unique RevCommit mergeBase = walk.next();
-      LOG.debug(() -> "Detected merge base for ${c1.getHash().getHashString()} " +
-          "and ${c2.getHash().getHashString()} is ${mergeBase.getId().getName()}");
+      @Nullable RevCommit mergeBase = walk.next();
```

Comment:
If I change that to `@Nullable RevCommit mergeBase = null;`, then compilation fails

Reply:
Looks like `@Nullable` here is ignored (?)